File: app/src/main/java/com/mss/imagesearcher/repositories/network/PixabayAPI.java
package com.mss.imagesearcher.repositories.network;

import androidx.annotation.NonNull;

import com.mss.imagesearcher.repositories.network.models.ImagesResponse;

import io.reactivex.Maybe;
import retrofit2.http.GET;
import retrofit2.http.Query;

public interface PixabayAPI {

    @NonNull
    @GET("api/")
    Maybe<ImagesResponse> findImages(@NonNull final @Query("key") String key,
                                     @NonNull final @Query("q") String query,
                                     @NonNull final @Query("lang") String lang,
                                     @NonNull final @Query("image_type") String imageType,
                                     @Query("page") int pageNumber,
                                     @Query("per_page") int imagesPerPage);
}

File: app/src/main/java/com/mss/imagesearcher/repositories/network/PixabayAPI.kt
package com.mss.imagesearcher.repositories.network

import com.mss.imagesearcher.repositories.network.models.ImagesResponse

import io.reactivex.Maybe
import retrofit2.http.GET
import retrofit2.http.Query

interface PixabayAPI {

    @GET("api/")
    fun findImages(@Query("key") key: String,
                   @Query("q") query: String,
                   @Query("lang") lang: String,
                   @Query("image_type") imageType: String,
                   @Query("page") pageNumber: Int,
                   @Query("per_page") imagesPerPage: Int): Maybe<ImagesResponse>
}
